src/api/app/models/package_file.rb

```diff
@@ -6,6 +6,6 @@ class PackageFile < Backend::File
 
   # calculates the real url on the backend to search the file
   def full_path(query = {})
-    URI.encode("/source/#{project_name}/#{package_name}/#{name}") + "?#{query.to_query}"
+    CGI.unescape("/source/#{project_name}/#{package_name}/#{name}") + "?#{query.to_query}"
```

Comment:
Unescape? Don't you mean escape?

Reply:
Yes, we know that's not right. We're fixing it [somewhere else][1] to reduce this PR. 

[1]: #12304